clang/include/clang/Driver/CC1Options.td

```diff
@@ -392,6 +392,8 @@ def header_include_file : Separate<["-"], "header-include-file">,
   HelpText<"Filename (or -) to write header include output to">;
 def show_includes : Flag<["--"], "show-includes">,
   HelpText<"Print cl.exe style /showIncludes to stdout">;
+def dependency_filter : Separate<["-"], "dependency-filter">,
+  HelpText<"Filter dependencies with prefix from the dependency output.">;
```

Comment:
Simple string comparison is used to filter out entries in the dependency file. The string specified in the -dependency-filter option is checked for prefix match.  Without the dependency-filter option, the .d file contains an entry inc/foo.h.  Any substring: i in inc inc/ .. inc/foo.h will match and filter out the entry.

Reply:
So the match happens against a path (`inc/foo.h`), not a filename component of a path (`foo.h`)? Relative to what directory this path has to be specified (current directory or a file's directory)?

Does it matter if I change the example as follows (include a header with a relative path and without `-I` option):

```
$ ls 
inc/foo.h
foo.cpp

$ cat foo.cpp
#include "inc/foo.h"

$ clang++ foo.cpp
```

Reply:
In this case, the .d file shows file.o depends on inc/foo.h.  We expect that this is an expert option, and the user who wants to filter the dependency file understands how to format the prefix string. If I needed to use this option, I would build the project using the -MD option and then study the .d files created by the compiler to see how the filter string should be set.  As this is a cc1 option, it's not exposed in the compiler documentation, and we expect that primarily it will be used by the compiler writer.